### DISCUSSION TITLE
Using GraphEdge.fill

### DISCUSSION THREAD

**Discussion by faseidl** [reaviz/reagraph#293] [Q&A]
I need to set the color of individual edges, but I cannot get GraphEdge.fill to have an effect. Is there something I must do on GraphCanvas to get it to use the fill color specified with each edge?

**Comment by quang-alethea** [comment]
I am seeing this case also. Does anyone know how to change edge colors?

**Comment by amcdnl** [comment]
Sounds like a bug - can you create a issue with an example test case and we can fix it.

**Reply by Nerivec** [reply]
Somewhat related to https://github.com/reaviz/reagraph/issues/268
The problem appears to be with the merged mesh code path (`animated: false`), it then only uses the theme colors.
https://github.com/reaviz/reagraph/blob/a75d06385952ac78aaaa3f7aea7a938b62669fcb/src/GraphScene.tsx#L495-L508
https://github.com/reaviz/reagraph/blob/a75d06385952ac78aaaa3f7aea7a938b62669fcb/src/symbols/edges/Edges.tsx#L233-L250

I don't think three would support separate coloring with the `BufferGeometry`, except maybe using far more advanced material application?
Maybe an option on GraphScene to [override the codepath](https://github.com/reaviz/reagraph/blob/a75d06385952ac78aaaa3f7aea7a938b62669fcb/src/GraphScene.tsx#L476-L477) even if `animated: false` would be an alternative until a better solution comes along (with a quick note that this is at the cost of performance optimization - but for smaller graphs, should be fine)?
